data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OTelProtoStandardCodec.java

```diff
@@ -328,16 +324,15 @@ private Map<String, Object> getStatus(final io.opentelemetry.proto.trace.v1.Stat
 
         protected Span parseSpan(final io.opentelemetry.proto.trace.v1.Span sp, final Map<String, Object> instrumentationScopeAttributes,
                                     final String serviceName, final Map<String, Object> resourceAttributes, final Instant timeReceived) {
-            return JacksonStandardSpan.builder()
+            Span span = JacksonStandardSpan.builder()
                     .withSpanId(convertByteStringToString(sp.getSpanId()))
                     .withTraceId(convertByteStringToString(sp.getTraceId()))
                     .withTraceState(sp.getTraceState())
                     .withParentSpanId(convertByteStringToString(sp.getParentSpanId()))
                     .withName(sp.getName())
-                    .withServiceName(serviceName)
```

Comment:
If you create a trace group, why do you remove the service name? Both are additional fields in the OpenSearch universe.

Reply:
Good point. I can put back service name for trace group. Thanks!

Reply:
I will put it as part of `otel_trace` processor, not by default. By default, it looks like OTEL logs/metrics/traces do not have something called "ServiceName". If the customers need it, it can be easily be added using "add_entries" processor by getting the value from Resource attributes.